### PR TITLE
Remove IActivityGenerator, change ActivityGenerator to ActivityFactory

### PR DESCRIPTION
**Pull request by cosmicshuai**
close: https://github.com/microsoft/botbuilder-dotnet/issues/2791

- remove IActivityGenerator class.

- change ActivityGenerator to ActivityFactory class

- remove Generate method and GeneratedFromLG method

- add CreateActivity static method which takes the result from Generate method in languageGenerator

- modify BindToData method in ActivityTemplate

### CODE REVIEW

File: libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/Generators/ActivityFactory.cs [hunk 1]
```diff
@@ -17,10 +17,10 @@
 namespace Microsoft.Bot.Builder.Dialogs.Adaptive.Generators
 {
     /// <summary>
-    /// The ActivityGenerator implements IActivityGenerator by using ILanguageGenerator
+    /// The ActivityFactory by using ILanguageGenerator
```
Comment (Danieladu) [inline libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/Generators/ActivityFactory.cs]: remove "by using ILanguageGenerator"

File: libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/Generators/ActivityFactory.cs [hunk 2]
```diff
@@ -71,7 +67,7 @@ public static Activity GenerateFromLG(object lgResult)
         /// <param name="template">(optional) inline template definition.</param>
         /// <param name="data">data to bind the template to.</param>
         /// <returns>activity.</returns>
-        public async Task<Activity> Generate(ITurnContext turnContext, string template, object data)
+        public static async Task<Activity> Generate(ITurnContext turnContext, string template, object data)
```
Comment (boydc2014) [inline libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/Generators/ActivityFactory.cs]: By removing IActivityGenerator, we want to remove this method also. 

This ActivityFactor should accept a string and return an object.

File: libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/LGAdapterExtensions.cs [hunk 3]
```diff
@@ -50,27 +50,24 @@ public static class LGAdapterExtensions
...
         }
 
         /// <summary>
         /// Register MessageActivityGeneration. 
         /// </summary>
         /// <param name="botAdapter">botAdapter to add services to.</param>
-        /// <param name="messageGenerator">(OPTIONAL) Default is TextMessageActivityGenerator(). </param>
         /// <returns>botAdapter.</returns>
-        public static BotAdapter UseMessageActivityGeneration(this BotAdapter botAdapter, IActivityGenerator messageGenerator = null)
+        public static BotAdapter UseMessageActivityGeneration(this BotAdapter botAdapter)
```
Comment (boydc2014) [inline libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/LGAdapterExtensions.cs]: Do we still need this?

File: libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/Microsoft.Bot.Builder.Dialogs.Adaptive.csproj [hunk 4]
```diff
@@ -61,7 +61,6 @@
   </ItemGroup>
 
   <ItemGroup>
-    <Folder Include="Generators\" />
```
Comment (boydc2014) [inline libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/Microsoft.Bot.Builder.Dialogs.Adaptive.csproj]: Why this?

**Comment by fuselabs** [comment]
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.dll** compared against [version 4.5.3](https://www.nuget.org/packages/Microsoft.Bot.Builder/4.5.3)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.AI.Luis.dll** compared against [version 4.5.3](https://www.nuget.org/packages/Microsoft.Bot.Builder.AI.Luis/4.5.3)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.AI.QnA.dll** compared against [version 4.5.3](https://www.nuget.org/packages/Microsoft.Bot.Builder.AI.QnA/4.5.3)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.ApplicationInsights.dll** compared against [version 4.5.3](https://www.nuget.org/packages/Microsoft.Bot.Builder.ApplicationInsights/4.5.3)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.Azure.dll** compared against [version 4.5.3](https://www.nuget.org/packages/Microsoft.Bot.Builder.Azure/4.5.3)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.Dialogs.dll** compared against [version 4.5.3](https://www.nuget.org/packages/Microsoft.Bot.Builder.Dialogs/4.5.3)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.TemplateManager.dll** compared against [version 4.5.3](https://www.nuget.org/packages/Microsoft.Bot.Builder.TemplateManager/4.5.3)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Configuration.dll** compared against [version 4.5.3](https://www.nuget.org/packages/Microsoft.Bot.Configuration/4.5.3)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Connector.dll** compared against [version 4.5.3](https://www.nuget.org/packages/Microsoft.Bot.Connector/4.5.3)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Schema.dll** compared against [version 4.5.3](https://www.nuget.org/packages/Microsoft.Bot.Schema/4.5.3)

File: libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/Templates/ActivityTemplate.cs [hunk 5]
```diff
@@ -34,36 +35,11 @@ public virtual async Task<Activity> BindToData(ITurnContext context, object data
...
-                        data: data).ConfigureAwait(false);
-                    if (result != null)
-                    {
-                        message.Text = result;
-                        message.Speak = result;
-                    }
-                }
-
-                return message as Activity;
+                var result = await ActivityFactory.Generate(
```
Comment (boydc2014) [inline libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/Templates/ActivityTemplate.cs]: No, in here, you should call

ActivityFactory.CreateActivity(ILanguageGenerator.Generate(...)).

**Comment by boydc2014** [comment]
> #2791
> 
> * remove IActivityGenerator class.
> * change ActivityGenerator to ActivityFactory class
> * generate method now is a static method

Also please use github keyword, like closes #xxx, to make sure the issue gets closed after this is merged.

File: tests/Microsoft.Bot.Builder.Dialogs.Adaptive.Templates.Tests/MessageGeneratorTests.cs [hunk 6]
```diff
@@ -49,72 +51,89 @@ public async Task TestInlineActivityGenerator()
...
             AssertActivityWithHeroCardAttachment(activity);
 
-            activity = await mg.Generate(context, "[activityWithMultiAttachments]", data: data) as Activity;
+            lgStringResult = await languageGenerator.Generate(context, "[activityWithMultiAttachments]", data: data).ConfigureAwait(false);
+            activity = ActivityFactory.CreateActivity(lgStringResult) as Activity;
             AssertActivityWithMultiAttachments(activity);
 
-            activity = await mg.Generate(context, "[activityWithSuggestionActions]", data: data) as Activity;
+            lgStringResult = await languageGenerator.Generate(context, "[activityWithSuggestionActions]", data: data).ConfigureAwait(false);
+            activity = ActivityFactory.CreateActivity(lgStringResult) as Activity;
```
Comment (boydc2014) [inline tests/Microsoft.Bot.Builder.Dialogs.Adaptive.Templates.Tests/MessageGeneratorTests.cs]: why you have to use "As", i tink it's Activity, right?

File: tests/Microsoft.Bot.Builder.Dialogs.Adaptive.Templates.Tests/MessageGeneratorTests.cs [hunk 7]
```diff
@@ -128,60 +147,60 @@ public void TestGenerateFromLG()
...
-            activity = ActivityGenerator.GenerateFromLG(engine.EvaluateTemplate("activityWithHeroCardAttachment", data));
+            activity = ActivityFactory.GenerateFromLG(engine.EvaluateTemplate("activityWithHeroCardAttachment", data));
             AssertActivityWithHeroCardAttachment(activity);
 
-            activity = ActivityGenerator.GenerateFromLG(engine.EvaluateTemplate("activityWithMultiAttachments", data));
+            activity = ActivityFactory.GenerateFromLG(engine.EvaluateTemplate("activityWithMultiAttachments", data));
             AssertActivityWithMultiAttachments(activity);
 
-            activity = ActivityGenerator.GenerateFromLG(engine.EvaluateTemplate("activityWithSuggestionActions", data));
+            activity = ActivityFactory.GenerateFromLG(engine.EvaluateTemplate("activityWithSuggestionActions", data));
```
Comment (boydc2014) [inline tests/Microsoft.Bot.Builder.Dialogs.Adaptive.Templates.Tests/MessageGeneratorTests.cs]: Where this generateFromLG come from, i think we don't have it now?

**Comment by fuselabs** [comment]
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.dll** compared against [version 4.5.3](https://www.nuget.org/packages/Microsoft.Bot.Builder/4.5.3)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.AI.Luis.dll** compared against [version 4.5.3](https://www.nuget.org/packages/Microsoft.Bot.Builder.AI.Luis/4.5.3)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.AI.QnA.dll** compared against [version 4.5.3](https://www.nuget.org/packages/Microsoft.Bot.Builder.AI.QnA/4.5.3)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.ApplicationInsights.dll** compared against [version 4.5.3](https://www.nuget.org/packages/Microsoft.Bot.Builder.ApplicationInsights/4.5.3)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.Azure.dll** compared against [version 4.5.3](https://www.nuget.org/packages/Microsoft.Bot.Builder.Azure/4.5.3)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.Dialogs.dll** compared against [version 4.5.3](https://www.nuget.org/packages/Microsoft.Bot.Builder.Dialogs/4.5.3)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.TemplateManager.dll** compared against [version 4.5.3](https://www.nuget.org/packages/Microsoft.Bot.Builder.TemplateManager/4.5.3)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Configuration.dll** compared against [version 4.5.3](https://www.nuget.org/packages/Microsoft.Bot.Configuration/4.5.3)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Connector.dll** compared against [version 4.5.3](https://www.nuget.org/packages/Microsoft.Bot.Connector/4.5.3)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Schema.dll** compared against [version 4.5.3](https://www.nuget.org/packages/Microsoft.Bot.Schema/4.5.3)

**Pull request merged**